modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1466,6 +1468,70 @@ public void onError(@NotNull Throwable e) {
                 LOG.error("Error while processing pending assignments event", e);
             }
         });
+
+        metaStorageMgr.registerWatchByPrefix(ByteArray.fromString(STABLE_ASSIGNMENTS_PREFIX), new WatchListener() {
+            @Override
+            public boolean onUpdate(@NotNull WatchEvent evt) {
+                if (!busyLock.enterBusy()) {
+                    throw new IgniteInternalException(new NodeStoppingException());
+                }
+
+                try {
+                    assert evt.single();
+
+                    if (evt.entryEvent().newEntry().value() == null) {
+                        return true;
+                    }
+
+                    int part = extractPartitionNumber(evt.entryEvent().newEntry().key());
+                    UUID tblId = extractTableId(evt.entryEvent().newEntry().key(), STABLE_ASSIGNMENTS_PREFIX);
+
+                    var stableAssignments = metaStorageMgr.get(partAssignmentsStableKey(partitionRaftGroupName(tblId, part))).join();
+
+                    if (!Arrays.equals(stableAssignments.value(), evt.entryEvent().newEntry().value())) {
+                        return true;
+                    }
+
+                    TableImpl tbl = tablesByIdVv.latest().get(tblId);
+
+                    ExtendedTableConfiguration tblCfg = (ExtendedTableConfiguration) tablesCfg.tables().get(tbl.name());
+
+                    String grpId = partitionRaftGroupName(tblId, part);
+
+                    List<List<ClusterNode>> assignments = (List<List<ClusterNode>>)
+                            ByteUtils.fromBytes(tblCfg.assignments().value());
+
+                    List<ClusterNode> newPeers = ((List<ClusterNode>) ByteUtils.fromBytes(evt.entryEvent().newEntry().value()));
+
+                    var toStop = assignments.get(part).stream()
+                            .filter(p -> !newPeers.contains(p))
+                            .collect(Collectors.toList());
+
+                    try {
+                        ClusterNode localMember = raftMgr.server().clusterService().topologyService().localMember();
+
+                        if (toStop.contains(localMember)) {
+                            raftMgr.stopRaftGroup(grpId);
+
+                            RaftGroupService partGrpSvc = tbl.internalTable().partitionRaftGroupService(part);
+
+                            partGrpSvc.shutdown();
```

Comment:
As I know, we have `RaftGroupService` for all partitions on every node to process requests for any partition on any node. So, we shouldn't stop it here.